src/hipscat/catalog/catalog.py

```diff
@@ -73,6 +75,22 @@ def filter_by_cone(self, ra: float, dec: float, radius: float) -> Catalog:
         )
         return Catalog(filtered_catalog_info, filtered_cone_pixels)
 
+    def filter_by_polygon(self, polygon: PolygonSkyRegion) -> Tuple[Catalog, int]:
+        """Filter the pixels in the catalog to only include the pixels that overlap
+        with a polygonal sky region
+
+        Args:
+            polygon (PolygonSkyRegion): The polygon, represented by spherical
+                sky coordinates (ra and dec), to filter points with
+
+        Returns:
+            A new catalog with only the pixels that overlap with the specified
+            polygon, and their maximum pixel order
```

Comment:
Yeah, I think that's a good idea. It can be an attribute method that wraps the `max(pixel_tree.pixels.keys())` call that happens elsewhere.

Even without that though, I'm not sure the max_order is needed here though. I see that it's used in the LSDB PR for this, but there it's used to generate the polygon to perform the filtering at the point level, and that should really be using a higher order than the max order here, or ideally using astropy to get it completely accurate. I'll review that PR too and explain more there.

Reply:
Yes, @smcguire-cmu, you're right! We don't need the `max_order` here anymore. Also, following our discussions, I replaced the usage of astropy `PolygonSkyRegion` with [spherical-geometry](https://github.com/spacetelescope/spherical_geometry) `SingleSphericalPolygon` .